### PR TITLE
Supermatter surge now asks for severity + should we announce when admin triggered

### LINKED ISSUE #11181: Supermatter Surge event does not ask if it should be announced
If I force trigger the event it forces the event.

Also, no option for severity?!!

### PR DESCRIPTION
**Pull request by Bibby0110**
# Github documenting your Pull Request

Fixes https://github.com/yogstation13/Yogstation/issues/11181

# Wiki Documentation

# Changelog

admin abuse deserves no changelog

### CODE REVIEW

File: code/modules/events/supermatter_surge.dm [hunk 1]
```diff
@@ -4,18 +4,36 @@
...
 	if(GLOB.main_supermatter_engine?.has_been_powered)
 		return ..()
 
 /datum/round_event/supermatter_surge
 	var/power = 2000
+	announceWhen = 1
 
 /datum/round_event/supermatter_surge/setup()
-	power = rand(1000,100000)
+	var/datum/round_event_control/falsealarm/C = control
```
Comment (adamsong) [inline code/modules/events/supermatter_surge.dm]:
```suggestion
	var/datum/round_event_control/supermatter_surge/C = control
```

Pretty sure this is the correct type

**Review by adamsong — COMMENTED** [review]
I don't see where it is giving the option to choose if it is announced

**Comment by Bibby0110** [comment]
> I don't see where it is giving the option to choose if it is announced

Handled elsewhere, requires announce_when  to be above 0 which it wasn't by default

**Comment by Xoxeyos** [comment]
Could we please also get an administration log, when an admin forces it?

File: code/modules/events/supermatter_surge.dm [hunk 2]
```diff
@@ -4,18 +4,36 @@
...
 	max_occurrences = 4
 	earliest_start = 10 MINUTES
+	var/forced_power
+
+/datum/round_event_control/supermatter_surge/admin_setup()
+	if(!check_rights(R_FUN))
+		return
+
+	var/selected_power = input("How severe should it be? Between 1000 and 100000") as num|null
+	if(!selected_power)
```
Comment (adamsong) [inline code/modules/events/supermatter_surge.dm:14]: This should revert to random, not set the value to minmum

File: code/modules/events/supermatter_surge.dm [hunk 3]
```diff
@@ -4,18 +4,36 @@
 	weight = 15
 	max_occurrences = 4
 	earliest_start = 10 MINUTES
+	var/forced_power
+
+/datum/round_event_control/supermatter_surge/admin_setup()
+	if(!check_rights(R_FUN))
+		return
+
+	var/selected_power = input("How severe should it be? Between 1000 and 100000") as num|null
```
Comment (adamsong) [inline code/modules/events/supermatter_surge.dm]: You might want to make the class, not raw value, its the more familiar number

File: code/modules/events/supermatter_surge.dm [hunk 4]
```diff
@@ -4,18 +4,36 @@
...
+	if(!selected_power)
+		selected_power = 1000
+	selected_power = clamp(selected_power, 1000, 100000)
```
Comment (adamsong) [inline code/modules/events/supermatter_surge.dm:16]: Why clamp it? The adminbus demands class -10000

**Review by adamsong — CHANGES_REQUESTED** [review]

File: code/modules/events/supermatter_surge.dm (hunk 4 above)
Comment (Bibby0110) [reply code/modules/events/supermatter_surge.dm:16]: It's clamped, deal with it

File: code/modules/events/supermatter_surge.dm (hunk 3 above)
Comment (Bibby0110) [inline code/modules/events/supermatter_surge.dm]:
```suggestion
	var/selected_power = input("How severe should it be? Between 1000 and 100000. Press cancel for random") as num|null
```

File: code/modules/events/supermatter_surge.dm [hunk 5]
```diff
@@ -4,18 +4,36 @@
...
 	earliest_start = 10 MINUTES
+	var/forced_power
+
+/datum/round_event_control/supermatter_surge/admin_setup()
+	if(!check_rights(R_FUN))
+		return
+
+	var/selected_power = input("How severe should it be? Between 1000 and 100000") as num|null
+	if(!selected_power)
+		selected_power = 1000
```
Comment (Bibby0110) [inline code/modules/events/supermatter_surge.dm]:
```suggestion
		selected_power = rand(1000, 100000)
```

**Comment by Bibby0110** [comment]
> Could we please also get an administration log, when an admin forces it?

Same logs as forcing other random events, what more do you need?

**Review by adamsong — APPROVED** [review]
Still think it shouldn't be clamped but this will do

**Pull request merged**